locust/runners.py

```diff
@@ -440,6 +449,12 @@ def heartbeat_worker(self):
                 else:
                     client.heartbeat -= 1
 
+            if self.worker_count - len(self.clients.missing) <= 0:
+                self.stop(info='No workers remaining, stopping test.')
```

Comment:
Unless I'm mistaken, this check and call to `stop()` should be able to go into the if-block within the for loop above (immediately after `client.user_count = 0` at line 448):

```python
if client.heartbeat < 0 and client.state != STATE_MISSING:
```

We could then first log something like `The last worker went missing, stopping test.` and we wouldn't need to pass in `info` to stop.

Reply:
This change fixed the broken test case!